src/libraries/System.Security.Cryptography.Xml/tests/ReferenceTest.cs

```diff
@@ -151,21 +159,31 @@ public void LoadXsltTransforms()
         public void LoadAllTransforms()
         {
             string test = "<Reference xmlns=\"http://www.w3.org/2000/09/xmldsig#\"><Transforms><Transform Algorithm=\"http://www.w3.org/2000/09/xmldsig#base64\" /><Transform Algorithm=\"http://www.w3.org/TR/2001/REC-xml-c14n-20010315\" /><Transform Algorithm=\"http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments\" /><Transform Algorithm=\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\" /><Transform Algorithm=\"http://www.w3.org/TR/1999/REC-xpath-19991116\"><XPath /></Transform>";
-            test += "<Transform Algorithm=\"http://www.w3.org/TR/1999/REC-xslt-19991116\">";
-            test += "<xsl:stylesheet xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" xmlns=\"http://www.w3.org/TR/xhtml1/strict\" exclude-result-prefixes=\"foo\" version=\"1.0\">";
-            test += "<xsl:output encoding=\"UTF-8\" indent=\"no\" method=\"xml\" />";
-            test += "<xsl:template match=\"/\"><html><head><title>Notaries</title>";
-            test += "</head><body><table><xsl:for-each select=\"Notaries/Notary\">";
-            test += "<tr><th><xsl:value-of select=\"@name\" /></th></tr></xsl:for-each>";
-            test += "</table></body></html></xsl:template></xsl:stylesheet></Transform>";
+            string xsltTransform = "<Transform Algorithm=\"http://www.w3.org/TR/1999/REC-xslt-19991116\">";
+            xsltTransform += "<xsl:stylesheet xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" xmlns=\"http://www.w3.org/TR/xhtml1/strict\" exclude-result-prefixes=\"foo\" version=\"1.0\">";
+            xsltTransform += "<xsl:output encoding=\"UTF-8\" indent=\"no\" method=\"xml\" />";
+            xsltTransform += "<xsl:template match=\"/\"><html><head><title>Notaries</title>";
+            xsltTransform += "</head><body><table><xsl:for-each select=\"Notaries/Notary\">";
+            xsltTransform += "<tr><th><xsl:value-of select=\"@name\" /></th></tr></xsl:for-each>";
+            xsltTransform += "</table></body></html></xsl:template></xsl:stylesheet></Transform>";
+            int expectedTransformsCount = 6;
+#if NETCOREAPP
+            if (!RuntimeFeature.IsDynamicCodeSupported)
+            {
+                // XSLTs are only supported when dynamic code is supported. See https://github.com/dotnet/runtime/issues/84389
```

Comment:
I was curious to know how this test was passing before. `XmlDsigXsltTransform` uses `XslCompiledTransform`, which only operates using `DynamicMethods` (it is in the name - "Compiled").

But it looks like this test only _loads_ the XML into transforms, it doesn't actually transform it. So that's how it was passing before - it didn't go through the full pipeline.

Reply:
hmm is it actually a problem if we block using this completely now? can we move the check to when we're actually running a transform?

Reply:
One reason I cut it off here is to cut out all this "dead code" when using ASP.NET's DataProtection. Moving it would add a little more dead code, but at least the XslCompiledTransform class would be trimmed.

I'm not sure why someone would just load the XML without actually performing the transforms. I think we can wait for customer feedback here. I don't think this is going to be used much with applications that can't IL Ref.Emit.

Reply:
ah ok, then my refactoring probably broke that trimming, will put the method back